### PR TITLE
DOC-2029: Added API docs for setEditableRoot/hasEditableRoot

### PR DESCRIPTION
**Pull request by spocke**
Related Ticket: DOC-2029

Description of Changes:
* Added API docs for the new `setEditableRoot` and `hasEditableRoot` functions

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

**Review by vpyshnenko — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/api/Editor.ts [hunk 1]
```diff
@@ -1080,10 +1080,22 @@ class Editor implements EditorObservable {
     VisualAids.addVisual(this, elm);
   }
 
+  /**
+   * Changes the editable state of the editor root element.
```
Comment (brian-forte) [inline modules/tinymce/src/core/main/ts/api/Editor.ts:1084]:
```suggestion
   * Changes the editable state of the editor’s root element.
```

Inflect for the possessive case with the Saxon genitive.

File: modules/tinymce/src/core/main/ts/api/Editor.ts [hunk 2]
```diff
@@ -1080,10 +1080,22 @@ class Editor implements EditorObservable {
...
+   */
   public setEditableRoot(state: boolean): void {
     EditableRoot.setEditableRoot(this, state);
   }
 
+  /**
+   * Returns the current editable state of the editor root element.
+   *
+   * @method hasEditableRoot
+   * @return {Boolean} True if the root element is editable, false if it's not editable.
```
Comment (brian-forte) [inline modules/tinymce/src/core/main/ts/api/Editor.ts:1097]:
```suggestion
   * Returns the current editable state of the editor’s root element.
   *
   * @method hasEditableRoot
   * @return {Boolean} True if the root element is editable, false if it is not editable.
```
1. Inflect for the possessive case with the Saxon genitive.
2. expand _it’s_ to _it is_ for formal clarity.

File: modules/tinymce/src/core/main/ts/api/Editor.ts (hunk 1 above)
Comment (spocke) [reply modules/tinymce/src/core/main/ts/api/Editor.ts]: Changed this. Thanks for noticing this.

File: modules/tinymce/src/core/main/ts/api/Editor.ts (hunk 2 above)
Comment (spocke) [reply modules/tinymce/src/core/main/ts/api/Editor.ts]: Makes sense. Changed this.

**Review by brian-forte — APPROVED** [review]

**Pull request merged**